src/util/tab-events.js

```diff
@@ -1,5 +1,20 @@
 import eventToPromise from './event-to-promise'
 
+function tabChanged(tabId) {
+    return new Promise((resolve, reject) => {
+        browser.webNavigation.onCommitted.addListener((details)=>{
+            console.log(details);
+            if(details.tabId == tabId && details.frameId == 0)
+                reject();
```

Comment:
this might cause a double rejection since line 13 does the same thing. Even if this is allowed by the A+ pattern (the second rejection would be ignored) I wonder if there's any point in having a listener working but the promise rejected this way.